spec/models/reservation_spec.rb

```diff
@@ -87,705 +73,830 @@
-      expect(reservation.approved?).to be_falsey
+  describe '#renew' do
+    let!(:user) { FactoryGirl.build_stubbed(:user) }
+    it "doesn't renew if ineligible" do
+      res = FactoryGirl.build_stubbed(:valid_reservation)
+      allow(res).to receive(:eligible_for_renew?).and_return(false)
+      expect(res.renew(user)).to match(/not eligible/)
+    end
+    context 'eligible for renew' do
+      let!(:res) do
+        FactoryGirl.build_stubbed(:valid_reservation,
+                                  times_renewed: 0,
+                                  notes: '')
+      end
+      let!(:new_date) { Time.zone.today + 5.days }
+      before do
+        allow(res).to receive(:eligible_for_renew?).and_return(true)
+        allow(res).to receive(:find_renewal_date).and_return(new_date)
+        allow(res).to receive(:save)
+      end
+      it 'updates the due date' do
+        old_date = res.due_date
+        expect { res.renew(user) }.to change { res.due_date }
+          .from(old_date).to(new_date)
+      end
+      it 'increments times_renewed' do
+        expect { res.renew(user) }.to change { res.times_renewed }.by(1)
+      end
+      it 'updates the notes' do
+        time_string = Time.zone.now.to_s(:long)
+        due_date_string = new_date.to_s(:long)
+        allow(user).to receive(:md_link).and_return('md link')
+        res.renew(user)
+        expect(res.notes).to match(time_string)
+        expect(res.notes).to match(due_date_string)
+        expect(res.notes).to match(/md link/)
+        expect(res.notes).to match(/Renewed/)
+      end
+      it 'returns nil when successful' do
+        allow(res).to receive(:save).and_return(true)
+        expect(res.renew(user)).to be_nil
+      end
+      it 'returns error if unsuccessful' do
+        allow(res).to receive(:save).and_return(false)
+        expect(res.renew(user)).to be_kind_of(String)
+      end
     end
+  end
 
-    it 'returns false if denied' do
-      reservation.update_attributes(status: 'denied')
-      expect(reservation.approved?).to be_falsey
+  describe '#checkin' do
+    let!(:handler) { FactoryGirl.build_stubbed(:checkout_person) }
+    context 'no procedures' do
+      let!(:res) { FactoryGirl.build_stubbed(:checked_out_reservation) }
+      it 'sets the checkin handler' do
+        expect { res.checkin(handler, {}, '') }.to \
+          change { res.checkin_handler }.from(nil).to(handler)
+      end
+      it 'sets the checked in time' do
+        # travel in order to freeze the time
+        travel(-1.days) do
+          expect { res.checkin(handler, {}, '') }.to \
+            change { res.checked_in }.from(nil).to(Time.zone.now)
+        end
+      end
+      it 'sets the status to returned' do
+        expect { res.checkin(handler, {}, '') }.to \
+          change { res.status }.from('checked_out').to('returned')
+      end
+      it 'updates the notes' do
+        # FIXME: might fall under overtesting (testing messages sent to self)
```

Comment:
Hmm yea I think we'll just handle it in a separate object so we'll just change the subject of the `allow` an `expect` calls.